app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  validates :email, presence: true, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
+  validates :email, uniqueness: { case_sensitive: false }
+  validates :password_digest, presence: true
```

Comment:
Not sure if we can do this validation and the `has_secure_password` or if this is the right way to have this type of validation... 
The `has_secure_password` validator does **ALOT**, so it might be doing this already. I would double-check both on [Rails docs](https://api.rubyonrails.org/classes/ActiveModel/SecurePassword/ClassMethods.html) and the bcrypt docs/lesson. I remember something specific about how this validation has to work to stay secure and not cause in-app issues. 

I'll look into it more too. Obviously, I need to refresh my memory also!

Reply:
When I wrote my rails app in mod 3 this is how I created it. I basically used that project to show me how to set it up. Let's have a discussion about the specifics you are wanting to know. I was under the impression that we were mostly interested in getting OAuth-type verification set up for this long run and this was just to get the app running for now. If you have specific concerns, we can definitely look into it more.

Reply:
corrected model validation